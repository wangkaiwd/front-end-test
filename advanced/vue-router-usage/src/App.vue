<template>
  <div id="app">
    <router-link :to="{name: 'clothing'}">服装</router-link>
    <br>
    <router-link :to="{name: 'men-wear'}">男装</router-link>
    -
    <router-link :to="{name: 'shirt'}">衬衫</router-link>
    <br>
    <router-link :to="{name: 'women-wear'}">女装</router-link>
    -
    <router-link :to="{name: 'skirt'}">裙子</router-link>
    <br>
    <router-link :to="{name: 'liquor'}">酒类</router-link>
    <br>
    <router-link :to="{name: 'medicine'}">药类</router-link>
    <router-view/>
  </div>
</template>

<style>
  #app {
    font-family: Avenir, Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    text-align: center;
    color: #2c3e50;
  }

  #nav {
    padding: 30px;
  }

  #nav a {
    font-weight: bold;
    color: #2c3e50;
  }

  #nav a.router-link-exact-active {
    color: #42b983;
  }
</style>
