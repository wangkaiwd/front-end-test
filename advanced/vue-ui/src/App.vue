<template>
  <div id="app">
    <lt-icon></lt-icon>
    <lt-button></lt-button>
  </div>
</template>

<script>
  export default {
    name: 'App',
    components: {}
  };
</script>

<style lang="scss">
  #app {
  }
</style>
