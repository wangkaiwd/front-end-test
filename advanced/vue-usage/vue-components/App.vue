<template>
  <div>
    App
  </div>
</template>

<script>
  export default {
    data () {
      return {
        msg: '1'
      };
    }
  };
</script>

<style scoped>

</style>
