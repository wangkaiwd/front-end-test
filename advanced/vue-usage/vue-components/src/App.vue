<template>
  <div id="app">
    <!--    <communicate></communicate>-->
    <!--    <my-form></my-form>-->
    <my-menu></my-menu>
  </div>
</template>

<script>
  import communicate from '@/view/communicate';
  import MyMenu from '@/view/my-menu';
  import MyForm from '@/view/my-form';

  export default {
    name: 'App',
    components: {
      communicate,
      MyMenu,
      MyForm
    },
  };
</script>

<style lang="scss">
  #app {
  }
</style>
