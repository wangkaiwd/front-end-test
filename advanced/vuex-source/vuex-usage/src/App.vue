<template>
  <div id="app">
    <h2>
      root年龄{{$store.state.age}}
    </h2>
    <h3>
      <!--      a姓名{{$store.state.a.name}}-->
    </h3>
    <h3>
      <!--      b姓名{{$store.state.b.name}}-->
    </h3>
    <button @click="onAsyncAdd">async add</button>
    <button @click="onAdd">add</button>
    姓名{{$store.getters.personName}}
    {{age}}
    {{personName}}
  </div>
</template>

<script>
  import { mapGetters, mapMutations, mapState } from './store/my-vuex';

  export default {
    name: 'App',
    components: {},
    computed: {
      ...mapState(['age']),
      // age () { // 辅助函数是这种方式的简写
      //   return this.$store.state.age;
      // }
      ...mapGetters(['personName'])
    },
    data () {
      return {};
    },
    mounted () {
      // console.log('this.$store', this.$store);
    },
    methods: {
      ...mapMutations({ a: 'syncChange' }),
      onAsyncAdd () {
        this.$store.dispatch('asyncChange', 10);
      },
      onAdd () {
        // 没有设置命名空间将会继承父命名空间
        // 1. inherits the namespace from parent module
        // 2. further nest the namespace
        // this.$store.commit('syncChange', 10);
        this.a(10);
      }
    }
  };
</script>

<style lang="scss">
  #app {
    font-family: Avenir, Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    text-align: center;
    color: #2c3e50;
    margin-top: 60px;
  }
</style>
