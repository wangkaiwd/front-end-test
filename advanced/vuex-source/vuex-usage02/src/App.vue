<template>
  <div id="app">
    <div id="nav">
      <router-link to="/">Home</router-link>
      |
      <router-link to="/about">About</router-link>
    </div>
    <h3>a:{{ a }}</h3>
    <h3>c:{{ c }}</h3>
    <h2>age:{{ $store.state.d.age }}</h2>
    <button @click="addA">addA</button>
    <button @click="addC">addC</button>
    <button @click="changeAge">changeAge</button>
    <router-view/>
  </div>
</template>
<script>
export default {
  data () {
    return {};
  },
  mounted () {
    // 将Vue的实例放到了$store属性上
    console.log('$store', this.$store);
  },
  computed: {
    a () {
      return this.$store.state.a;
    },
    c () {
      return this.$store.state.b.c;
    }
  },
  methods: {
    addA () {
      this.$store.commit('addA');
    },
    addC () {
      // this.$store.commit('addC');
      this.$store.commit('b/addC');
    },
    changeAge () {
      this.$store.commit('changeAge', 10);
    }
  }
};
</script>
