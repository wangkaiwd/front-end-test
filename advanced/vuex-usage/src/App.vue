<template>
  <div id="app">
    <h2>
      root年龄{{$store.state.age}}
    </h2>
    <h3>
      <!--      a姓名{{$store.state.a.name}}-->
    </h3>
    <h3>
      <!--      b姓名{{$store.state.b.name}}-->
    </h3>
    <button @click="onAsyncAdd">async add</button>
    <button @click="onAdd">add</button>
    <button @click="$store.state.age = 100">click</button>
    姓名{{$store.getters.personName}}
  </div>
</template>

<script>

  export default {
    name: 'App',
    components: {},
    data () {
      return {};
    },
    mounted () {
      console.log(this.$store);
    },
    methods: {
      onAsyncAdd () {
        this.$store.dispatch('asyncChange', 10);
      },
      onAdd () {
        this.$store.commit('syncChange', 10);
      }
    }
  };
</script>

<style lang="scss">
  #app {
    font-family: Avenir, Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    text-align: center;
    color: #2c3e50;
    margin-top: 60px;
  }
</style>
