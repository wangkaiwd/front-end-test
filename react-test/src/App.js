import React from 'react';

function App () {
  return (
    // 可以使用data-test自定义属性来标记dom属性，通过enzyme来的api来进行选择
    <div className="App" title="test title" data-test="app">
      hello world
    </div>
  );
}

export default App;
