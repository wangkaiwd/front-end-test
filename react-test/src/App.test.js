import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

test('渲染测试', () => {
  // 只渲染App,对于App中的子组件只是以一个标记来代替，并不是真正渲染
  const wrapper = shallow(<App/>);
  // expect(wrapper.find('div').length).toBe(1);
  // 通过属性选择器，来保证即使代码的class(使用类选择器的话)发生改变，测试用例也可以照常使用
  // expect(wrapper.find('[data-test="app"]').prop('title')).toBe('test title');
  expect(wrapper.find('[data-test="app"]')).toExist();
});
