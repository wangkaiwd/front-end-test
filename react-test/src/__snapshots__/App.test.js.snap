// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染测试 1`] = `
<div
  className="App"
  data-test="app"
  title="test title"
>
  hello world
</div>
`;
